cypress/integration/slot_booking_bbmp.js

```diff
@@ -2,45 +2,46 @@ describe("My First Test", () => {
   it("Open Cowin App!", () => {
     cy.viewport(1000, 1000);
     cy.visit("https://selfregistration.cowin.gov.in/");
-    cy.get("input#mat-input-0").type("<your_mobile_number>");
+    cy.get("input#mat-input-0").type("9968726089");
     cy.get(".next-btn").click();
     cy.get("input#mat-input-1").focus();
-    cy.pause();
-
-    cy.get(".next-btn").click();
-
-    cy.intercept("*beneficiaries*").as("beneficiaries");
-    cy.wait(["@beneficiaries"]);
-
-    cy.get(".btnlist > .bordernone").first().click();
-
-    // TO Test
-    // cy.visit("https://selfregistration.cowin.gov.in/appointment");
-    cy.get(".custom-checkbox").click();
-
-    cy.get(".mat-form-field-infix").first().click();
-    cy.get("#mat-option-16").click();
-
-    cy.wait(100);
-
-    // cy.intercept("*districts*").as("districts");
-    // cy.wait(["@districts"]);
-
-    cy.get(".mat-form-field-infix").last().click();
-    cy.get("#mat-option-40").click();
-
-    cy.get(".pin-search-btn").click();
-
-    // cy.intercept("*calendarByDistrict*").as("calendarByDistrict");
-    // cy.wait(["@calendarByDistrict"]);
-
-    cy.get(".agefilterblock .form-check").first().click();
-
-    const slot = cy.get(".slots-box").not(".no-seat").not(".no-available");
-    if (slot) {
-      slot.first().click();
-    } else {
-      console.log("Not Available");
-    }
+    cy.pause();    
   });
+  /*
+  * 1. User needs to click on Verify button 
+  * 2. User needs to select beneficiaries by clicking on schedule 
+  * 3. Click Cypress Resume
+  * 4. Below method will rerun untill it will find a slot 
+  * 5. Need to add some delay 
+  */
+  it('.select() - select an option in a <select> element', 
+  {
+    "retries": 200000
+  },() => {    
+    // https://on.cypress.io/select   
+    
+    cy.get('.custom-checkbox').click('right')
```

Comment:
This is really nice.
For some reason in consecutive attempts the test is failing here:
```
cy.get('[formcontrolname="state_id"]').click("right");
``` 

So, I think we can do a something like this:
```
    it("Search by District", () => {
      cy.get(".custom-checkbox").click("right");
    });

    it("Select BBMP", () => {
      cy.get('[formcontrolname="state_id"]').click();
      cy.wait(100);
      cy.get('[role="listbox"]').contains("Karnataka").click();
      cy.wait(200);
      cy.get('[formcontrolname="district_id"]').click();
      cy.wait(100);
      cy.get('[role="listbox"]').contains("BBMP").click();
      cy.wait(200);
    });

   it(
      ".select() - select an option in a <select> element",
      {
        retries: 200000,
      },
      () => {
             ....
             ....
      });
)}
```

Thoughts??

Reply:
yeah this makes more sense. we need to click on search & find slots in each run. One more thing it logoff after some runs so we need to run these retries only if url is appointment.